doc/data-warehouse-schema.md

```diff
@@ -0,0 +1,311 @@
+storing metrics at a content item level, but at a base_path level (page level).
+This is due to the existence of some content items, like guides, that share the
+same content id for muliple pages.
+
+Please, read the following ADRs to get additional information.
+
+1. [Track collected metrics by base_path][adr-track-metrics-by-base-path]
+1. [Track attribute changes by base_path][adr-track-content-changes-per-base-path]
+1. [Convention for unique warehoue identifier][adr-warehouse-item-id]
+
+#### Dates dimension
+
+```
+Contains date attributes that are used to add time context to metrics and
+editions.
+```
+
+All metrics and editions needs time to be complete, because:
+
+1. Each metric has a **foreign key to the day** the meassure was collected.
+1. Each edition has a **foreign key to the day** the edition was created.
+
+You can also observe that we have other attributes in the dimension (year,
+week, month...), these attributes are used to enable grouping and aggregations;
+This is a common practices in data warehouses.
+
+### Aggregations
+
+ Aggregations are a common practice in data warehouses. There is no simpler way
+ to speed up queries than aggregating data.
+
+Usually aggregations are applied to facts tables. In our case, we are creating
+monthly aggregations for the facts metrics table.
+
+#### Monthly aggregations ####
+
+There are two tables involved in montly aggregations:
+
+##### Monthly metrics: table `monthly_metrics` #####
+
+```
+Stores the sum of all aggregable metrics in a month. It also stores the
+aggregation for the current month.
+```
+
+On a daily basis, the system will update the aggregations for the current month,
+creating a new entry at the beginning of each month.
+
+##### Month dimension: table `month_dimensions` #####
+
+```
+Contains all the months to current day. It is updated automatically each month.
+```
+
+We reference each metric in the aggregations table to the month in which it was
+collected (via a foreighn key to the monthly dimensions table).
+
+##### Benefits  #####
+
+The improvement in performance is substantial. Let's see the difference in
+nunber of rows performing a simple calculation: Total number of unique pageviews
+in a year for HMRC:
+
+
+* **With no aggregation**:
+
+```
+80K Editions x 365 days = 29.200.000 rows
+```
+
+* **With no aggregation**:
+
+```
+80K Editions x 12 days = 960.000  rows
+```
+
+The aggregations is decreasing the query time in around 30 times. For example,
+if the query on the facts table takes 60 seconds, running the same query on the
+aggregated table would take around 2 seconds.
+
+##### Drawbacks
+
+We should not create aggregations just to improve a query. Aggregations add
+complexity to the data warehouse and can be a source for bugs. Before creating
+new aggregations we would need to think about:
+
+1. How frequently do we use the Aggregation.
+1. Who is the user of the information. Is he able to wait a bit more if he only
```

Comment:
```suggestion
1. Who is the user of the information? Is longer query latency acceptable for queries that only run occasionally?
```

Reply:
Side note: try to avoid gendered pronouns/nouns. e.g "Is he able to wait" -> "Are they able to wait"